root/static/scripts/edit/URLCleanup.js

```diff
@@ -5087,6 +5090,88 @@ const CLEANUPS: CleanupEntries = {
+    restrict: [
+      LINK_TYPES.youtubemusic,
+      {
+        recording: LINK_TYPES.streamingfree.recording,
+        release: LINK_TYPES.streamingfree.release,
+      },
+      {
+        recording: LINK_TYPES.streamingpaid.recording,
+        release: LINK_TYPES.streamingpaid.release,
+      },
+    ],
+    clean: function (url) {
+      url = url.replace(/^(https?:\/\/)?music\.youtube\.com(?:\/#)?/, 'https://music.youtube.com');
+      // Channel (artist) URL
+      url = url.replace(/\/channel\/([^\/?#]+).*$/, '/channel/$1');
+      // Video (track) URL
+      url = url.replace(/^https:\/\/music\.youtube\.com\/.*[?&](v=[a-zA-Z0-9_-]+).*$/, 'https://music.youtube.com/watch?$1');
+      // Playlist (release) URL
+      url = url.replace(/^https:\/\/music\.youtube\.com\/playlist.*[?&](list=[a-zA-Z0-9_-]+).*$/, 'https://music.youtube.com/playlist?$1');
+      return url;
+    },
+    validate: function (url, id) {
+      if (/https:\/\/music\.youtube\.com\/search\?/.test(url)) {
+        return {
+          error: noLinkToSearchMsg(),
+          result: false,
+          target: ERROR_TARGETS.URL,
+        };
+      }
+      switch (id) {
+        case LINK_TYPES.youtubemusic.artist:
+          if (/^https:\/\/www\.youtube\.com\/playlist\?list=[a-zA-Z0-9_-]+/.test(url)) {
+            return {
+              error: l(
+                `This is a release (playlist) link, not an artist channel.
+                 Please link to the artist channel instead.`,
+              ),
+              result: false,
+              target: ERROR_TARGETS.ENTITY,
+            };
+          }
+          if (/^https:\/\/www\.youtube\.com\/watch\?v=[a-zA-Z0-9_-]+$/.test(url)) {
+            return {
+              error: linkToChannelMsg(),
+              result: false,
+              target: ERROR_TARGETS.ENTITY,
+            };
+          }
+          return {
+            result: /^https:\/\/music\.youtube\.com\/channel\/[^\/?#]+$/.test(url),
+            target: ERROR_TARGETS.URL,
+          };
+        case LINK_TYPES.streamingfree.recording:
+        case LINK_TYPES.streamingpaid.recording:
+          if (/^https:\/\/music\.youtube\.com\/watch\?v=[a-zA-Z0-9_-]+$/.test(url)) {
+            return {result: true};
+          }
+          return {
+            error: linkToVideoMsg(),
+            result: false,
+            target: ERROR_TARGETS.ENTITY,
+          };
+        case LINK_TYPES.streamingfree.release:
+        case LINK_TYPES.streamingpaid.release:
+          if (/^https:\/\/music\.youtube\.com\/playlist\?list=[a-zA-Z0-9_-]+$/.test(url)) {
+            return {result: true};
+          }
+          return {
+            error: l(
+              `Please link to a specific release (playlist) link. Add track
+               (video) links to the relevant recordings and channel links
+               to the relevant artists instead.`,
+            ),
+            result: false,
+            target: ERROR_TARGETS.ENTITY,
+          };
```

Comment:
Someone pasting a `browse` URL such as `https://music.youtube.com/browse/MPREb_UaQXOH57k99` would be prompted this message about track video and recordings which doesn’t seem appropriate.

Reply:
Hmm. Ok, that's interesting, I didn't realize `/browse` could be a redirect to `/playlist`. Wonder if it should be explicitly blocked as a redirect? Is it always a redirect? This thing is confusing.

Reply:
I don’t know if there is a general rule, I just found it while searching for potential label pages.

Reply:
Split this error into three, with the `/browse` block being an `ERROR_TARGETS.URL` error instead - does that seem better?

I did notice that all the album links in the https://music.youtube.com frontpage are `/browse` links redirecting to `/playlist` ones so this is probably very common. It doesn't seem like the `/browse` link ID and the `/playlist` link ID are related in any way, so we can't do a smarter thing.

Reply:
Not sure what have been the code changes in the meantime, but at least an appropriate message is now returned about the link being redirect, which seems to be the smartest thing to do indeed. :clap: